campfire/ui/src/components/ShareFileDialog.tsx

```diff
@@ -0,0 +1,355 @@
+
+        }
+
+        setFiles(undefined)
+
+        if (inputRef.current)
+            inputRef.current.value = '';
+
+        setTabOpen('Files')
+
+    }
+
+    return (
+        <div className="overflow-hidden rounded-xl">
+            <Flex className=" h-min relative  px-8 "
+                flexDirection='column'
+                gap={16}
+                style={{
+                    minHeight: '500px',
+                    maxHeight: 'min(80vh, 800px)',
+                    width: '500px',
+                    maxWidth: '500px',
+                    overflowY: 'auto',
+                    overflowX: 'hidden',
+                }}
+            >
+                <Flex className='w-full sticky py-4 pb-2 z-10 top-0 baseColor' justifyContent='center' alignItems='center'>
+                    <Card >
+                        <Button className='whitespace-nowrap' variant={tabOpen == 'Share' ? 'primary' : 'secondary'}
+                            onClick={() => setTabOpen('Share')}
+                            style={{ width: '130px' }}
+                        >
+                            Share Files
+                        </Button>
+                        <Button className='whitespace-nowrap' variant={tabOpen == 'Share' ? 'secondary' : 'primary'}
+                            onClick={() => setTabOpen('Files')}
+                            style={{ width: '130px' }}
+                        >
+                            All Files
+                        </Button>
+                    </Card>
+                </Flex>
+                {tabOpen === 'Share' ?
+                    <Flex flexDirection='column' className=' w-full h-full' gap={16}>
+                        <div style={{ borderRadius: '8px' }} className={` ${!(files === undefined || files.length === 0) && 'default-ring-always'}`}>
+                            <Card>
+                                <Flex flexDirection='column'
+                                    className='w-full h-full relative py-14'
+                                    justifyContent='center'
+                                    alignItems='center'
+                                >
+                                    {
+                                        (files === undefined || files.length === 0) &&
+                                        <>
+                                            <Text >Drop files here</Text>
+                                            <Text >or</Text>
+                                            <Text fontWeight='medium'>Click to browse</Text>
+                                        </>
+
+                                    }
+                                    {
+                                        files && files.length === 1 &&
+                                        <>
+                                            <MdUploadFile className="w-8 h-8 mb-2" style={{ color: 'var(--rlm-icon-color, #000000)', opacity: '0.5', fontSize: '20px' }} aria-hidden />
```

Comment:
nice job using the `var(--rlm-icon-color, #000000)`. Thats the code that makes realm theming work